packages/components/forms/src/Multiselect/Multiselect.tsx

```diff
@@ -0,0 +1,286 @@
+    // https://stackoverflow.com/a/46012210/17269164
+
+    const nativeInputValueSetter = Object.getOwnPropertyDescriptor(
+      window.HTMLInputElement.prototype,
+      'value',
+    ).set;
+    nativeInputValueSetter.call(internalSearchInputRef.current, '');
+    const forcedEvent = new Event('change', { bubbles: true });
+    internalSearchInputRef.current.dispatchEvent(forcedEvent);
+  };
+
+  const renderMultiselectLabel = React.useCallback(() => {
+    if (currentSelection.length === 0) {
+      return <>{placeholder}</>;
+    }
+    const leftoverCount = currentSelection.length - 1;
+    if (leftoverCount === 0) {
+      return (
+        <span
+          data-test-id="cf-multiselect-current-selection"
+          className={styles.currentSelection}
+        >
+          {currentSelection[0]}
+        </span>
+      );
+    }
+    return (
+      <span
+        data-test-id="cf-multiselect-current-selection"
+        className={styles.currentSelection}
+      >
+        {currentSelection[0]} and {leftoverCount} more{' '}
+      </span>
+    );
+  }, [currentSelection, placeholder, styles.currentSelection]);
+
+  const childrenLength = React.Children.count(children);
+
+  return (
+    <div
+      data-test-id={testId}
+      className={cx(styles.multiselect, className)}
+      ref={ref}
+    >
+      <Popover
+        usePortal={usePortal}
+        isOpen={isOpen}
+        onClose={() => setIsOpen(false)}
+        isFullWidth={listWidth === 'full'}
+        renderOnlyWhenOpen={false}
+      >
+        <Popover.Trigger>
+          <Button
+            aria-label="Toggle Multiselect"
+            ref={toggleRef}
+            onClick={() => setIsOpen(!isOpen)}
+            startIcon={startIcon}
+            endIcon={<ChevronDownIcon />}
+          >
+            {renderMultiselectLabel()}
+          </Button>
+        </Popover.Trigger>
+        <Popover.Content
+          ref={listRef}
+          className={styles.content(listMaxHeight)}
+          testId="cf-multiselect-container"
+        >
+          <>
+            {hasSearch && (
+              <>
+                <TextInput
+                  aria-label="Search"
+                  type="text"
+                  value={searchValue}
+                  className={styles.inputField}
+                  id={id}
+                  data-test-id="cf-multiselect-search"
+                  placeholder={searchPlaceholder}
+                  onChange={(event) => {
+                    handleSearchChange(event);
+                  }}
+                  ref={mergeRefs(searchInputRef, internalSearchInputRef)}
+                />
+                <IconButton
+                  aria-label="Clear"
+                  className={styles.toggleButton}
+                  variant="transparent"
+                  icon={
+                    searchValue ? (
+                      <CloseIcon variant="muted" />
```

Comment:
Should these icons be configurable?

Reply:
As far as i talked with @FerdinandoDeMeo only the startIcon should be configurable, so the icons inside the search input shall be magnify glass and the close one.
But if needed, we can update those to be configurable too in the future.